### ISSUE TITLE
MouseDragging does not register left mouse button for in a child window

### ISSUE THREAD

**Issue by Noremos** [ocornut/imgui#6213]
**Version/Branch of Dear ImGui:**

Version: Dear ImGui 1.89.4 WIP (18933)
Branch: _(master/docking)_

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Compiler: MSBuild (Visual Studio)
Operating System: Windows 11

**My Issue/Question:**

It seems like ImGui::IsMouseDragging(ImGuiMouseButton_Left) is not working for a child. It works fine with the ImGuiMouseButton_Right flag.

**Screenshots/Video**

https://user-images.githubusercontent.com/29798473/222377866-c0fd1ad8-a70e-4214-8102-6194089c8c83.mp4

```Cpp
{
    ImGui::SetNextWindowPos(ImVec2(10, 10));
    ImGui::SetNextWindowSize(ImVec2(200, 300));
    ImGui::Begin("Main");

    if (ImGui::BeginChild("Child", ImVec2(100,100)))
    {
        auto* list = ImGui::GetWindowDrawList();
        auto start = ImGui::GetCursorPos() + ImGui::GetWindowPos();
        list->AddRectFilled(start, start + ImVec2(100, 100), ImColor(200,0,0));
    }
    ImGui::EndChild();

    if (ImGui::IsWindowHovered() || ImGui::IsItemHovered())
    {
        if (ImGui::IsMouseDragging(ImGuiMouseButton_Left))
            ImGui::Text("Left");

        if (ImGui::IsMouseDragging(ImGuiMouseButton_Right))
            ImGui::Text("Right");
    }

    ImGui::End();
 }

``````

**Comment by GamingMinds-DanielC** [comment]
Do you need it to be specifically a child? If not, a properly sized and manually drawn `ImGui::InvisibleButton()` might be an option. With fitting flags for the mouse buttons you can then use its active status, similar to the dragging example from the demo window.

**Comment by Noremos** [comment]
> Do you need it to be specifically a child? If not, a properly sized and manually drawn `ImGui::InvisibleButton()` might be an option. With fitting flags for the mouse buttons you can then use its active status, similar to the dragging example from the demo window.

I will try your option. Thanks

**Comment by ocornut** [comment]
> It seems like ImGui::IsMouseDragging(ImGuiMouseButton_Left) is not working for a child. It works fine with the ImGuiMouseButton_Right flag.

You issue seemlingly has nothing to do with `IsMouseDragging()`. If you display raw state:

```
ImGui::EndChild();
ImGui::Text("IsItemHovered %d", ImGui::IsItemHovered());
ImGui::Text("IsWindowHovered %d", ImGui::IsWindowHovered());

ImGui::Text("IsMouseDragging Left %d", ImGui::IsMouseDragging(ImGuiMouseButton_Left));
ImGui::Text("IsMouseDragging Right %d", ImGui::IsMouseDragging(ImGuiMouseButton_Right));
```

You'll notice `IsMouseDragging()` works and your test was filtered by your call to `IsItemHovered()`.

TL;DR; your `if (ImGui::IsWindowHovered() || ImGui::IsItemHovered())` branch is probably incorrect.

But it's hard to provide a correct answer without knowing your actual use case.
When you click the "void" section of a window it takes active id to move the parent window (which here is overriden by the `SetNextWindowPos()` call). It is normal that many "hovered" calls tends to return false by default when another item is active. Please refer to `ImGuiHoveredFlags flags` for `IsItemHovered()`. Or you may need to submit an item using InvisibleButton() as suggest, but it depends what you are trying to do.

**Comment by Noremos** [comment]
> Please refer to `ImGuiHoveredFlags flags`

That is what I need! Thanks. ImGuiHoveredFlags_ChildWindows solves the issue. But it's still pretty confusinп that in the same case ImGuiMouseButton_Right works fine, but ImGuiMouseButton_Left doesn't.

**Comment by ocornut** [comment]
Because left buttons attempts to drag the window. Right button doesn’t.
